main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/ziputil"
+	"github.com/bitrise-steplib/steps-appetize-io-deploy/appetize"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+	"github.com/bitrise-tools/go-steputils/tools"
+)
+
+// Config ...
+type Config struct {
+	AppPath   string          `env:"app_path,required"`
+	Platform  string          `env:"platform,opt[ios,android]"`
+	Token     stepconf.Secret `env:"appetize_token,required"`
+	PublicKey string          `env:"public_key"`
+	Verbose   bool            `env:"verbose,required"`
+}
+
+func main() {
+	var cfg Config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Issue with input: %s", err)
+	}
+
+	log.SetEnableDebugLog(cfg.Verbose)
+
+	stepconf.Print(cfg)
+	fmt.Println()
+
+	log.Infof("Checking provided file extension")
+	var pth string
+	var err error
+	if pth, err = ensureExtension(cfg.AppPath, cfg.Platform); err != nil {
+		failf("Upload failed!\nError: %s", err)
+	}
+
+	fmt.Println()
+	log.Infof("Upload")
+	// curl https://tok_vjbxr9m95cwe7r3fpjpqh3y94w@api.appetize.io/v1/apps -F "file=@/Users/birmachera/Desktop/ip/XcodeArchiveTest.ipa" -F "platform=ios"
+	client := appetize.NewClient(string(cfg.Token), pth, cfg.Platform, cfg.PublicKey)
+
+	if cfg.PublicKey == "" {
+		log.Warnf("No public key provided  🚨")
+		log.Printf("Uploading new app to Appetize.io")
+	} else {
+		log.Printf("Public key provided: %s  ✅", cfg.PublicKey)
+		log.Printf("Updating the provided app at Appetize.io")
+	}
+
+	var response appetize.Response
+	if response, err = client.DirectFileUpload(); err != nil {
+		failf("Upload failed %s", err)
+	}
+
+	fmt.Println()
+	log.Printf("Upload succeeded  🎉")
+
+	logDebugPretty(&response)
+
+	appURL := path.Join("https://appetize.io/app", response.PublicKey)
+	log.Printf("You can check your app at: %s", appURL)
+	fmt.Println()
+
+	log.Infof("Generating output")
+
+	if err := tools.ExportEnvironmentWithEnvman("APPETIZE_APP_URL", appURL); err != nil {
+		failf("Failed to generate output - %s", "APPETIZE_APP_URL")
+	}
+
+	log.Donef("APPETIZE_APP_URL: %s", appURL)
+	fmt.Println()
+	log.Donef("Done")
+}
+
+func failf(format string, v ...interface{}) {
+	log.Errorf(format, v...)
+	log.Warnf("For more details you can enable the debug logs by turning on the verbose step input.")
+	os.Exit(1)
+}
+
+func logDebugPretty(v interface{}) {
+	b, err := json.MarshalIndent(v, "", "  ")
```

Comment:
this shall not even run if debug mode is off. please check debug mode here as well and skip marshalling as well